components/export/api/src/main/java/org/phenotips/export/internal/DataToCellConverter.java

```diff
@@ -283,100 +287,222 @@ public DataSection genesHeader() throws Exception
+                if (!present.contains(field)) {
+                    continue;
                 }
-            }
-        }
-        if (present.contains("rejected")) {
-            PatientData<Map<String, String>> rejectedGenes = patient.getData("rejectedGenes");
-            if (rejectedGenes != null && rejectedGenes.isIndexed()) {
-                DataCell cell = new DataCell("Previously tested", 0, y, StyleOption.YES_NO_SEPARATOR);
-                cell.addStyle(StyleOption.NO);
-                section.addCell(cell);
-                for (Map<String, String> rejectedGene : rejectedGenes) {
-                    hasGenes = true;
-                    String geneName = rejectedGene.get("gene");
-                    cell = new DataCell(geneName, 1, y);
-                    section.addCell(cell);
 
-                    if (present.contains("rejected_comments")) {
-                        String comment = rejectedGene.get("comments");
-                        cell = new DataCell(comment, 2, y);
-                        section.addCell(cell);
+                String value = variant.get(field);
+
+                if (translatables.contains(field)) {
+                    String translated = "";
+                    if (upperCase.contains(value)) {
+                        translated = capitalizeValue(value);
+                    } else {
+                        translated = valueTranslates.get(value);
                     }
-                    y++;
+                    cell = new DataCell(translated, j++, y);
+                } else if ("evidence".equals(field)) {
+                    cell = new DataCell(parseEvidence(value), j++, y);
+                } else {
+                    cell = new DataCell(value, j++, y);
                 }
+
+                section.addCell(cell);
             }
+            y++;
         }
 
-        /* Creating empties */
-        if (!hasGenes) {
-            /* Status, and gene name columns are always present */
-            int columns = present.contains("comments") || present.contains("rejected_comments") ? 3 : 2;
+        return section;
+    }
+
+    public DataSection genesBody(Patient patient) throws Exception
+    {
+        String sectionName = "genes";
+        Set<String> present = this.enabledHeaderIdsBySection.get(sectionName);
+        if (present == null || present.isEmpty()) {
+            return null;
+        }
+
+        DataSection section = new DataSection();
+        int y = 0;
+
+        PatientData<Map<String, String>> allGenes = patient.getData("genes");
+
+        // empties should be created in the case that there are no genes to write
+        if (allGenes == null || !allGenes.isIndexed()) {
+            /* Status and gene name columns are always present */
             Integer emptyX = 0;
-            for (int i = 0; i < columns; i++) {
-                DataCell cell = new DataCell("", emptyX, 0);
+            for (int i = 0; i < present.size(); i++) {
```

Comment:
you don't use `i` at all here. Maybe you should use it instead of `emptyX`, since they are identical.